File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque
def solution(priorities, location):
    queue = deque((idx, pro) for idx, pro in enumerate(priorities))
    answer = 0

    while True:
        current = queue.popleft()
        if any(current[1] < q[1] for q in queue):
            queue.append(current)
        else:
            answer += 1
            if current[0] == location:
                return answer

File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    var answer = 0;
    let queue = priorities.map((v, i) => [v, i]);
    
    while(queue){
        let popElement = queue.shift();
        if(queue.some(v => v[0] > popElement[0])){
            queue.push(popElement);
        } else {
            answer++;
            if(popElement[1] === location) return answer;
        }
    }
    return answer;
}